lib/route.js

```diff
@@ -213,7 +213,8 @@ function buildRouting (options) {
         opts.logLevel,
         opts.logSerializers,
         opts.attachValidation,
-        this[kReplySerializerDefault]
+        this[kReplySerializerDefault],
+        this[kOptions].ajv.customOptions.errorFormatter
```

Comment:
I'm not sure it is needed to add this reference to the request context: this feature is not customizable per route, so it should not be necessary since the server instance should be already bound somewhere
(You may argue that `opts.errorHandler || this._errorHandler,` is the same, and you are right.. maybe we could optimize this too)

I say that after the performance gain by PR #2428 

I'll check it deeply to suggest an approach

Reply:
I did this via `Context` since the validation happens in the request response cycle and I didn't see another way elsewhere in the project (I'd be glad if you correct me), I think your point is valid, and I'd be more than happy for advice on how to handle this.